corehq/apps/users/permissions.py

```diff
@@ -52,16 +52,10 @@ def can_upload_data_files(domain, couch_user):
 
 
 def _has_data_file_permission(domain, couch_user, read_only=True):
-    from corehq.apps.users.models import DomainMembershipError
```

Comment:
why does the domain check no longer matter?

Reply:
My hypothesis is it doesn't need a stricter domain check than typical permissions checks, the [original PR](https://github.com/dimagi/commcare-hq/pull/27108) doesn't mention anything unusual.

@snopoke is there a reason this is specifically limited to domain members? There's now a feature that lets users share permissions across domains without being members of all those domains, so the role check here is preventing them from using the dropzone.

Reply:
I'm going to merge this to unblock the project, but happy to do a followup PR if needed.

Reply:
> There's now a feature that lets users share permissions across domains without being members of all those domains

I wasn't aware of this, won't this mean that any user with the `can_edit_file_dropzone` permission can see any dropzone from any domain? Where does the domain check happen now?

Reply:
It's new, this PR: https://github.com/dimagi/commcare-hq/pull/27510

There's a configuration in django admin that maps which domains can access which. It's being used by covid projects that have a lot of domains (20-50).